src/csvcubedpmd/csvwcli/pull.py

```diff
@@ -134,20 +135,24 @@ def _get_rdf_file_dependencies(metadata_file: str) -> List[str]:
 
 
 def _get_table_group_for_metadata_file(metadata_file: str) -> Dict:
-    if isinstance(metadata_file, str):
+    if isinstance(metadata_file, str) and _looks_like_uri(metadata_file):
```

Comment:
Both of these conditions (the if and the else) require a string. If you want to enforce that use an assertion before calling the if block. i.e:

```python
assert isinstance(metadata_file, str), f"Unexpected type received: {type(metadata_file)"
``` 

Something to note - I wouldn't be surprised if you got a ValueError if you passed in say a boolean - the `_looks_like_uri(metadata_file)` would still run even though `isinstance(metadata_file, str) == False`. Always be a little careful with `and` statements, they're a common python gotcha.